### PR TITLE
update monolith id figure

### LINKED ISSUE #278: change `monolith-id-state` to be a sequence diagram

### PR DESCRIPTION
**Pull request by Victor-M-Giraldo**
closes #278

### CODE REVIEW

File: figures/monolith-id-sequence.mmd [hunk 1]
```diff
@@ -0,0 +1,13 @@
+sequenceDiagram
+
+    actor Jeff
```
Comment (dyc3) [inline figures/monolith-id-sequence.mmd:3]: remove jeff -- not sure why hes here

File: figures/monolith-id-sequence.mmd [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    Participant OTT
+    participant Monolith
+    participant Balancer
+
+    Note over OTT: Load Balancing Enabled
+    Jeff->>OTT: Goes to OTT
+    OTT->>Monolith: Client connects to Monolith
+    Note over Monolith: Generates ID
+    Monolith->>Balancer: Sends Init Message with Monolith ID
+    Note over Balancer: Uses ID to initialize connection
```
Comment (dyc3) [inline figures/monolith-id-sequence.mmd:13]:
```suggestion
    Note over Monolith,Balancer: Connection established
```

File: figures/monolith-id-sequence.mmd [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+    actor Jeff
+    Participant OTT
+    participant Monolith
+    participant Balancer
+
+    Note over OTT: Load Balancing Enabled
+    Jeff->>OTT: Goes to OTT
+    OTT->>Monolith: Client connects to Monolith
+    Note over Monolith: Generates ID
```
Comment (dyc3) [inline figures/monolith-id-sequence.mmd:11]: ID generation happens on startup

File: figures/monolith-id-sequence.mmd [hunk 4]
```diff
@@ -0,0 +1,13 @@
+sequenceDiagram
+
+    actor Jeff
+    Participant OTT
+    participant Monolith
+    participant Balancer
+
+    Note over OTT: Load Balancing Enabled
+    Jeff->>OTT: Goes to OTT
+    OTT->>Monolith: Client connects to Monolith
```
Comment (dyc3) [inline figures/monolith-id-sequence.mmd:10]: With the balancer, clients never connect to the monolith directly.

**Review by dyc3 — CHANGES_REQUESTED** [review]
This is completely nonsensical. The balancer connects to the monoliths, not the other way around. Clients are not relevant to this process.

File: figures/monolith-id-sequence.mmd (hunk 4 above)
Comment (Victor-M-Giraldo) [reply figures/monolith-id-sequence.mmd]: Do you just want to get rid of the Client part entirely?

File: figures/monolith-id-sequence.mmd [hunk 5]
```diff
@@ -1,13 +1,12 @@
...
     participant Monolith
     participant Balancer
 
     Note over OTT: Load Balancing Enabled
-    Jeff->>OTT: Goes to OTT
-    OTT->>Monolith: Client connects to Monolith
-    Note over Monolith: Generates ID
+
+    OTT->>Monolith: Client connects through Balancer
+    Note over Monolith: ID generated on startup
```
Comment (dyc3) [inline figures/monolith-id-sequence.mmd:10]: You need to change the diagram to reflect this information, not just change the labels

File: figures/monolith-id-sequence.mmd [hunk 6]
```diff
@@ -1,13 +1,12 @@
 sequenceDiagram
 
-    actor Jeff
     Participant OTT
```
Comment (dyc3) [inline figures/monolith-id-sequence.mmd:3]: This participant doesn't make any sense.

```suggestion
```

**Review by dyc3 — CHANGES_REQUESTED** [review]

**Comment by Victor-M-Giraldo** [comment]
Think this is better. Or should BalancerManager instead be a participator since that is what generates the UUID and sends it?

File: figures/monolith-id-sequence.mmd [hunk 7]
```diff
@@ -1,12 +1,8 @@
...
 
-    Note over OTT: Load Balancing Enabled
-
-    OTT->>Monolith: Client connects through Balancer
-    Note over Monolith: ID generated on startup
-    Monolith->>Balancer: Sends Init Message with Monolith ID
-    Note over Monolith,Balancer: Connection established
+    Monolith->>Monolith: Generate Monolith ID on startup
+    Monolith->>Balancer: Send Monolith ID in Init Message
+    Balancer->>Balancer: Receive MonolithId
```
Comment (dyc3) [inline figures/monolith-id-sequence.mmd:7]: This is implied

File: figures/monolith-id-sequence.mmd [hunk 8]
```diff
@@ -1,12 +1,8 @@
...
     participant Balancer
 
-    Note over OTT: Load Balancing Enabled
-
-    OTT->>Monolith: Client connects through Balancer
-    Note over Monolith: ID generated on startup
-    Monolith->>Balancer: Sends Init Message with Monolith ID
-    Note over Monolith,Balancer: Connection established
+    Monolith->>Monolith: Generate Monolith ID on startup
+    Monolith->>Balancer: Send Monolith ID in Init Message
```
Comment (dyc3) [inline figures/monolith-id-sequence.mmd:6]: Where's the connection step?

**Review by dyc3 — CHANGES_REQUESTED** [review]

File: figures/monolith-id-sequence.mmd [hunk 9]
```diff
@@ -0,0 +1,9 @@
+sequenceDiagram
+    participant Monolith
+    participant Balancer
+
+    Monolith->>Monolith: Generate Monolith ID on startup
+    Monolith->>Balancer: Send Monolith ID in Init Message
+    Balancer->>Monolith: Connects Balancer to Monolith
```
Comment (dyc3) [inline figures/monolith-id-sequence.mmd:7]: These are backwards.

**Review by dyc3 — CHANGES_REQUESTED** [review]

File: figures/monolith-id-sequence.mmd (hunk 9 above)
Comment (dyc3) [reply figures/monolith-id-sequence.mmd:7]: and the caption can just be "Connect"

Comment (Victor-M-Giraldo) [reply figures/monolith-id-sequence.mmd:7]: Thought they should be in that order since balancer connects to monolith. Changing now.

**Review by dyc3 — CHANGES_REQUESTED** [review]

**Comment by Victor-M-Giraldo** [comment]
I understand that the balancer connects to the monolith. In one of the previous reviews you commented where I had it as `Balancer->>Monolith: Connects Balancer to Monolith` that it was backwards, which is why I changed it back. My mistake.

The one thing I don't understand is why you want this to be `Balancer->>Monolith: Send Monolith ID in Init Message` rather than `Monolith->>Balancer: Send Monolith ID in Init Message`. Even in the related [issue](https://github.com/dyc3/opentogethertube/issues/1251), you describe it as the monolith generating the ID, and conveying it to the balancer in the init message.

The message type is also `MsgM2B` in `balancer.ts`.

Basically, why is it the other way around?

**Comment by dyc3** [comment]
No, you're misreading my comment. I left [this comment referring to the 2 lines here](https://github.com/dyc3/senior-design/pull/281#discussion_r1483647351). The order of operations is wrong.

It should be:
1. the balancer connects to the monolith
2. monolith sends init

The monolith cannot possibly send the init message if there is no connection.

**Comment by Victor-M-Giraldo** [comment]
Ok, that makes way more sense. I'll fix that now. Thanks.

**Review by dyc3 — APPROVED** [review]
